core/broadleaf-framework/src/main/java/org/broadleafcommerce/core/search/service/DatabaseSearchServiceImpl.java

```diff
@@ -17,113 +17,129 @@
+import org.broadleafcommerce.common.extension.ExtensionResultHolder;
+import org.broadleafcommerce.common.money.Money;
+import org.broadleafcommerce.common.web.BroadleafRequestContext;
 import org.broadleafcommerce.core.catalog.domain.Category;
 import org.broadleafcommerce.core.catalog.domain.Product;
 import org.broadleafcommerce.core.catalog.service.CatalogService;
 import org.broadleafcommerce.core.search.dao.FieldDao;
 import org.broadleafcommerce.core.search.dao.SearchFacetDao;
-import org.broadleafcommerce.core.search.domain.CategorySearchFacet;
-import org.broadleafcommerce.core.search.domain.Field;
-import org.broadleafcommerce.core.search.domain.FieldEntity;
-import org.broadleafcommerce.core.search.domain.SearchCriteria;
-import org.broadleafcommerce.core.search.domain.SearchFacet;
-import org.broadleafcommerce.core.search.domain.SearchFacetDTO;
-import org.broadleafcommerce.core.search.domain.SearchFacetRange;
-import org.broadleafcommerce.core.search.domain.SearchFacetResultDTO;
-import org.broadleafcommerce.core.search.domain.SearchResult;
+import org.broadleafcommerce.core.search.domain.*;
+import org.broadleafcommerce.core.search.extension.PriceListSearchFacetRangeExtensionManager;
 import org.broadleafcommerce.core.search.service.solr.SolrSearchServiceImpl;
+import org.springframework.beans.factory.annotation.Value;
 import org.springframework.stereotype.Service;
-import java.util.ArrayList;
-import java.util.Collections;
-import java.util.Comparator;
-import java.util.HashMap;
-import java.util.List;
-import java.util.Map;
-import java.util.Map.Entry;
-import javax.annotation.Resource;
 
-import net.sf.ehcache.Cache;
-import net.sf.ehcache.CacheManager;
-import net.sf.ehcache.Element;
+import javax.annotation.Resource;
+import javax.persistence.Tuple;
+import java.math.BigDecimal;
+import java.util.*;
+import java.util.Map.Entry;
 
 /**
- * @deprecated Use {@link SolrSearchServiceImpl} 
+ * @deprecated Use {@link SolrSearchServiceImpl}
  */
 @Deprecated
 @Service("blSearchService")
 public class DatabaseSearchServiceImpl implements SearchService {
-    
+
     @Resource(name = "blCatalogService")
     protected CatalogService catalogService;
-    
+
     @Resource(name = "blSearchFacetDao")
     protected SearchFacetDao searchFacetDao;
-    
+
     @Resource(name = "blFieldDao")
     protected FieldDao fieldDao;
-    
+
+    @Resource(name = "priceListSearchFacetRangeExtensionManager")
+    protected PriceListSearchFacetRangeExtensionManager extensionManager;
+
+    @Value("${database.searchfacets.enabled:true}")
+    protected boolean enabledFacets;
+
     protected static String CACHE_NAME = "blStandardElements";
     protected static String CACHE_KEY_PREFIX = "facet:";
     protected Cache cache = CacheManager.getInstance().getCache(CACHE_NAME);
-    
+
     @Override
     public SearchResult findExplicitSearchResultsByCategory(Category category, SearchCriteria searchCriteria) throws ServiceException {
```

Comment:
maybe that method should call findSearchResultsByCategory(Category category, SearchCriteria searchCriteria)  ?
Just an idea